packages/google_sign_in/google_sign_in_platform_interface/lib/google_sign_in_platform_interface.dart

```diff
@@ -86,7 +86,27 @@ abstract class GoogleSignInPlatform {
     String? hostedDomain,
     String? clientId,
   }) async {
-    throw UnimplementedError('init() has not been implemented.');
+    await initWithParams(
```

Comment:
This is unintuitive, but you have to do the delegation in the other direction: the new method has to call the old method, dropping any new params.

That way, when the app-facing package updates to use the new method, all existing implementations will continue to work because they will inherit the backward-compatible default implementation, which avoids breaking unendorsed implementations (and avoids the need to update the minimum allowed version of all of the implementation packages, since old implementation versions will also still work).

In implementations, on the other hand, you *do* delegate in this direction, so that you get the right behavior regardless of which is called.

Reply:
Gotcha, that makes sense. So the `MethodChannelGoogleSignIn` should go this direction (as it does), just not the interface, right?